File: 1915-check-if-one-string-swap-can-make-strings-equal/1915-check-if-one-string-swap-can-make-strings-equal.py
class Solution:
    def areAlmostEqual(self, s1: str, s2: str) -> bool:
        ind, cnt = -1, 0
        for i in range(len(s1)):
            if s1[i] != s2[i]:
                cnt += 1
                if ind == -1:
                    ind = i
                elif cnt > 2 or s1[ind] != s2[i] or s1[i] != s2[ind]:
                    return False
        return cnt != 1

File: 1915-check-if-one-string-swap-can-make-strings-equal/1915-check-if-one-string-swap-can-make-strings-equal.java
class Solution {
    public boolean areAlmostEqual(String s1, String s2) {
        int ind = -1,cnt = 0;
        for(int i = 0;i < s1.length();i++){
            if(s1.charAt(i) != s2.charAt(i)){
                cnt++;
                if(ind == -1)
                    ind = i;
                else if(cnt > 2 || s1.charAt(ind) != s2.charAt(i) || s1.charAt(i) != s2.charAt(ind))
                    return false;
            }
        }
        return (cnt != 1);
    }
}